### PR TITLE
[APM] link to related errors list in transaction details

### PR DESCRIPTION
**Pull request by ogupte**
Fixes #21920 by adding link to related errors list by filtering with the Kuery bar

transaction-view-related-errors

### CODE REVIEW

File: x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/StickyTransactionProperties.tsx [hunk 1]
```diff
@@ -91,5 +95,28 @@ export function StickyTransactionProperties({
...
+        <KibanaLink
+          pathname={'/app/apm'}
+          hash={`/${idx(transaction, _ => _.service.name)}/errors`}
+          query={{
+            kuery: `transaction.id~3A~22${transaction.transaction.id}~22`
+          }}
+        >
+          {i18n.translate('xpack.apm.transactionDetails.viewErrors', {
+            defaultMessage:
+              errorCount === 1 ? `View 1 Error` : `View ${errorCount} Errors`
```
Comment (ogupte) [inline x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/StickyTransactionProperties.tsx]: I'm not sure the best way to handle this translation with dynamic content in the text.

**Comment by ogupte** [comment]
The new `StickyTransactionPropertiesErrorRequest` component is a RRR wrapper for `StickyTransactionProperties` to provide `errorCount`. This way `StickyTransactionProperties` can still be used in other places (like the transaction flyout) without any changes to component props or performance impacts of the extra request.

File: x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/StickyTransactionProperties.tsx (hunk 1 above)
Comment (sqren) [reply x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/StickyTransactionProperties.tsx]:
```
defaultMessage: {errorCount, plural, one {View 1 error} other {View # errors}}
```

I've bookmarked this super helpful ICU editor: https://format-message.github.io/icu-message-format-for-translators/editor.html

File: x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/StickyTransactionProperties.tsx [hunk 2]
```diff
@@ -91,5 +95,28 @@ export function StickyTransactionProperties({
...
+    stickyProperties.push({
+      label: i18n.translate('xpack.apm.transactionDetails.errorsLabel', {
+        defaultMessage: 'Errors'
+      }),
+      val: (
+        <KibanaLink
+          pathname={'/app/apm'}
+          hash={`/${idx(transaction, _ => _.service.name)}/errors`}
+          query={{
+            kuery: `transaction.id~3A~22${transaction.transaction.id}~22`
```
Comment (sqren) [inline x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/StickyTransactionProperties.tsx]: Use `legacyEncodeURIComponent`:
```suggestion
            kuery: legacyEncodeURIComponent(`transaction.id : "${transaction.transaction.id}"`)
```

File: x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/StickyTransactionPropertiesErrorRequest.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+export const StickyTransactionPropertiesErrorRequest: React.SFC<Props> = ({
+  transaction,
+  urlParams: { errorGroupId, start, end },
+  ...restProps
+}) => {
+  return (
+    <ErrorGroupOverviewRequest
+      urlParams={{
```
Comment (sqren) [inline x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/StickyTransactionPropertiesErrorRequest.tsx]: Shouldn't this just be `urlParams={urlParams}`?

File: x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/StickyTransactionPropertiesErrorRequest.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        kuery: `transaction.id: "${transaction.transaction.id}"`,
+        serviceName: idx(transaction, _ => _.service.name),
```
Comment (sqren) [inline x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/StickyTransactionPropertiesErrorRequest.tsx]: Why are you not getting `serviceName` from `urlParams`?

**Comment by sqren** [comment]
@ogupte Using the `ErrorGroupOverviewRequest` is smart but I also think it's a bit of a hack. The intention is not to load a list of error groups. Rather it is to get the error count for a specific transaction.

To make this more clear, I think we should have backend method that does just that: given a `transactionId` and a `traceId` return the error count. 
Perhaps the `getTransaction` can call this method, so we don't have to make duplicate requests from the frontend. We already use `getTransaction` elsewhere, and we might not want get the error count every time we cat `getTransaction` so it could be a setting `get_error_count=true`. But that might be overengineering, since getting the error count should be super cheap. So I'm fine with doing it everytime for simplicity.

**Comment by jasonrhodes** [comment]
Just as a sidenote, it may not be _necessary_ to show the number of errors in this link, in which case we may not need a whole separate server-side method right now unless we find we need this error count in other places? "View related errors" seems like a perfectly fine link description here, but defer to @formgeist on how important that is etc. 

Just a thought :)

**Comment by formgeist** [comment]
Note from standup earlier: I think the number is valuable and we should show it to indicate to the user what to expect in the list when they've clicked the link. The second decision was to always show the cell, even if there's no errors associated with the transaction because that's also valuable information to the user. The cell will supply return `None` as value.

**Comment by ogupte** [comment]
I continued this work in a different branch with a PR that will merge to master https://github.com/elastic/kibana/pull/29313

**Pull request closed without merging**